src/DotNetStatus/Controllers/DataController.cs

```diff
@@ -0,0 +1,224 @@
+            };
+
+            return result;
+        }
+
+        private KeyValuePair<string, IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>>> BuildRollUpEnumerable(string key, IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>> value)
+        {
+            return new KeyValuePair<string, IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>>>(key, value);
+        }
+
+        private IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>> BuildRollUpKeyValuePair(BreakingChangeAnalyzerStatus breakingChangeAnalyzerStatus, int value)
+        {
+            return new List<KeyValuePair<BreakingChangeAnalyzerStatus, int>>
+            {
+                new KeyValuePair<BreakingChangeAnalyzerStatus, int>(breakingChangeAnalyzerStatus, value)
+            };
+        }
+
+        #endregion
+
+        #region Sorting
+        private IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>> SortKeyValuePair(
+            IEnumerable<KeyValuePair<BreakingChangeAnalyzerStatus, int>> enumToSort,
+            KeyValueSortFieldsEnum sortField)
+        {
+            if (sortField == KeyValueSortFieldsEnum.Value)
+            {
+                return (enumToSort.OrderBy(o => o.Value));
+            }
+            else
+            {
+                return (enumToSort.OrderBy(o => o.Key));
+            }
+        }
+
+        private IEnumerable<BreakingChange> SortBreakingChanges(IEnumerable<BreakingChange> enumToSort, BreakingChangeSortFieldEnum sortField)
+        {
+            int id;
+
+            switch (sortField)
+            {
+                case BreakingChangeSortFieldEnum.Id:
+                    return (enumToSort.OrderBy(o => int.TryParse(o.Id, out id) ? id : int.MaxValue));
+                case BreakingChangeSortFieldEnum.Title:
+                    return (enumToSort.OrderBy(o => o.Title));
+                default:
+                    return (enumToSort.OrderBy(o => int.TryParse(o.Id, out id) ? id : int.MaxValue));
+            }
+        }
+
+        private IEnumerable<DiagnosticAnalyzerInfo> SortDiagnosticAnalzerInfo(IEnumerable<DiagnosticAnalyzerInfo> enumToSort, AnalyzerInfoSortFieldEnum sortField)
+        {
+            switch (sortField)
+            {
+                case AnalyzerInfoSortFieldEnum.IdNumber:
+                    return (enumToSort.OrderBy(o => o.IdNumber));
+                case AnalyzerInfoSortFieldEnum.AnalyzerName:
+                    return (enumToSort.OrderBy(o => o.AnalyzerName));
+                default:
+                    return (enumToSort.OrderBy(o => o.IdNumber));
+            }
+        }
+    }
+
+    public enum AnalyzerInfoSortFieldEnum
+    {
+        IdNumber,
+        AnalyzerName
+    }
+
+    public enum BreakingChangeSortFieldEnum
+    {
+        Id,
+        Title
+    }
+
+    public enum KeyValueSortFieldsEnum
+    {
+        Key,
+        Value
+    }
```

Comment:
I removed the sorting altogether since this should really be done on the client not by the service. I just have default sorting now.